features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/instrumentation/gestures/GesturesListener.kt

```diff
@@ -122,13 +122,29 @@ internal class GesturesListener(
     // region Internal
 
     private fun closeScrollOrSwipeEventIfAny(decorView: View?, onUpEvent: MotionEvent) {
-        val type = scrollEventType ?: return
+        val type = scrollEventType
+        if (type == null) {
+            closeScrollAsTap(decorView, onUpEvent)
+        } else {
+            closeScrollOrSwipeEvent(type, decorView, onUpEvent)
+        }
+    }
+
+    private fun closeScrollAsTap(decorView: View?, onUpEvent: MotionEvent) {
+        if (decorView != null) {
+            val downTarget = findTargetForTap(decorView, onTouchDownXPos, onTouchDownYPos)
+            val upTarget = findTargetForTap(decorView, onUpEvent.x, onUpEvent.y)
 
+            if (downTarget == upTarget && downTarget != null) {
```

Comment:
I would suggest to use `===` operator here to highlight that we want exactly same instance by reference (because who knows, maybe it will be some `View` which redefines `equals`).